packages/bounds/src/index.ts

```diff
@@ -0,0 +1,154 @@
+export function getElementBounds(element: Element | undefined | null | false): NullableBounds {
+  if (!element)
+    return {
+      top: null,
+      left: null,
+      bottom: null,
+      right: null,
+      width: null,
+      height: null
+    };
+  return clientRectToBounds(element.getBoundingClientRect());
+}
+
+function clientRectToBounds(rect: {
+  top: number;
+  left: number;
+  bottom: number;
+  right: number;
+  width: number;
+  height: number;
+}): Bounds {
+  return {
+    top: rect.top,
+    left: rect.left,
+    bottom: rect.bottom,
+    right: rect.right,
+    width: rect.width,
+    height: rect.height
+  };
+}
+
+/**
+ * Creates a reactive store-like object of current element bounds — position on the screen, and size dimensions. Bounds will be automatically updated on scroll, resize events and updates to the dom.
+ *
+ * @param target Element for calculating bounds. Can be a reactive signal. Set to falsy value to disable tracking.
+ * @param options Choose which events should be tracked *(All are enabled by default)*
+ * - `options.trackScroll` — listen to window scroll events
+ * - `options.trackMutation` — listen to changes to the dom structure/styles
+ * - `options.trackResize` — listen to element's resize events
+ * @returns reactive object of {@link target} bounds
+ * @example
+ * ```ts
+ * const bounds = createElementBounds(document.querySelector("#my_elem")!);
+ *
+ * createEffect(() => {
+ *    console.log(
+ *      bounds.width, // => number
+ *      bounds.height, // => number
+ *      bounds.top, // => number
+ *      bounds.left, // => number
+ *      bounds.right, // => number
+ *      bounds.bottom, // => number
+ *    );
+ * });
+ * ```
+ */
+export function createElementBounds(
+  target: Accessor<Element> | Element,
+  options?: {
+    trackScroll?: boolean;
+    trackMutation?: boolean;
+    trackResize?: boolean;
+  }
+): Readonly<Bounds>;
+export function createElementBounds(
+  target: Accessor<Element | undefined | null | false> | Element,
+  options?: {
+    trackScroll?: boolean;
+    trackMutation?: boolean;
+    trackResize?: boolean;
+  }
+): Readonly<NullableBounds>;
+export function createElementBounds(
+  target: Accessor<Element | undefined | null | false> | Element,
+  {
+    trackMutation = true,
+    trackResize = true,
+    trackScroll = true
+  }: {
+    trackScroll?: boolean;
+    trackMutation?: boolean;
+    trackResize?: boolean;
+  } = {}
+): Readonly<NullableBounds> {
+  const [bounds, setBounds] = createStaticStore(getElementBounds(access(target)));
+  const updateBounds = () => setBounds(getElementBounds(access(target)));
+  const updateBoundsOf = (el: Element | undefined | null | false) =>
+    setBounds(getElementBounds(el));
+
+  if (typeof target === "function") {
+    onMount(() => updateBoundsOf(target()));
+    createComputed(on(target, updateBoundsOf, { defer: true }));
+  }
+
+  if (trackResize)
```

Comment:
By using a predefined options object, if the user tries `{ trackScroll: false }`, he'll be surprised that nothing will be tracked. It would be better to use an empty object by default and test whether `trackScroll` (or `scroll`, see earlier comment) was === false.

Reply:
I'm pretty sure it works properly...
The options are initialized as an empty object and destructured with some default values. So `{ trackScroll: false }` would only flip one switch.